rfcs/020 - Dependencies permissions RFC.md

```diff
@@ -51,21 +51,70 @@ This is the basic case when we want to use `new_dep_A` in our project but we do
 
 ![Block at library level](./lib-lvl-v1.png)
 
-The `.dependencies.ln` file could look something like:
-
-Block `@std/cmd` but do not override if any mock exists:
-```ln
-add('https://github.com/org/new_dep_A', some(['cmd']), false);
+The `@std/deps` will have the following API:
+
+```ts
+type Dependency {
+  url: string, // Empty string or ignore it for global blocks at application level
+  block: Array<string>,
+  fullBlock: Array<string>,  // fullBlock, deepBlock, extensiveBlock, exhaustiveBlock?
+}
+
+fn add(url: string) = new Dependency {
+  url: url,
+  block: [],
+  fullBlock: [],
+};
+
+fn add() = new Dependency {
+  url: '',
+  block: [],
+  fullBlock: [],
+};
+
+fn block(dep: Dependency, block: string): Dependency {
+  dep.block.push(block);
+  return dep;
+}
+
+fn block(dep: Dependency, block: Array<string>): Dependency {
+  dep.block = dep.block + block;
+  return dep;
+}
+
+fn fullBlock(dep: Dependency, block: string): Dependency {
+  dep.fullBlock.push(fullBlock);
+  return dep;
+}
+
+fn fullBlock(dep: Dependency, block: Array<string>): Dependency {
+  dep.fullBlock = dep.fullBlock + block;
+  return dep;
+}
```

Comment:
maybe instead of having the empty `add()` which serves no use besides constructing the `Dependency`, we can have `fn block(block: string) = new Dependency { ... }` and the same for `fullBlock`?

Reply:
Yeah probably thats more meaningful

Reply:
likewise, there should be an equivalent `fn block(block: Array<string>)` etc